base/commands/map/map_set.go

```diff
@@ -0,0 +1,67 @@
+//go:build base || map
+
+package _map
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hazelcast/hazelcast-go-client"
+
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/proto/codec"
+)
+
+type MapSetCommand struct{}
+
+func (mc *MapSetCommand) Init(cc plug.InitContext) error {
+	addKeyTypeFlag(cc)
+	addValueTypeFlag(cc)
+	cc.AddIntFlag(mapTTL, "", ttlUnset, false, "time-to-live (ms)")
+	cc.AddIntFlag(mapMaxIdle, "", ttlUnset, false, "max idle (ms)")
+	cc.SetPositionalArgCount(2, 2)
+	help := "Set a value in the given Map"
+	cc.SetCommandHelp(help, help)
+	cc.SetCommandUsage("set [-n MAP] KEY VALUE [flags]")
+	return nil
+}
+
+func (mc *MapSetCommand) Exec(ctx context.Context, ec plug.ExecContext) error {
+	mapName := ec.Props().GetString(mapFlagName)
+	ci, err := ec.ClientInternal(ctx)
+	if err != nil {
+		return err
+	}
+	// get the map just to ensure the corresponding proxy is created
+	if _, err := ec.Props().GetBlocking(mapPropertyName); err != nil {
+		return err
+	}
+	keyStr := ec.Args()[0]
+	valueStr := ec.Args()[1]
+	kd, vd, err := makeKeyValueData(ec, ci, keyStr, valueStr)
+	if err != nil {
+		return err
+	}
+	ttl := GetTTL(ec)
```

Comment:
When I check it ttl values and maxIdle values could not get correctly.

Reply:
Thanks for catching that, fixed at: https://github.com/hazelcast/hazelcast-commandline-client/pull/171/commits/bdd411ae442c264ab8ec898e973972356335353d